contracts/chain-adapters/Arbitrum_CustomGasToken_Adapter.sol

```diff
@@ -177,7 +187,7 @@ contract Arbitrum_CustomGasToken_Adapter is AdapterInterface, CircleCCTPAdapter
                 CUSTOM_GAS_TOKEN.safeIncreaseAllowance(address(L1_INBOX), amountToBridge);
                 L1_INBOX.createRetryableTicket(
                     to, // destAddr destination L2 contract address
-                    L2_CALL_VALUE, // l2CallValue call value for retryable L2 message
+                    _fromNativeTo18Decimals(amount), // l2CallValue call value for retryable L2 message
```

Comment:
This isn't just a decimal change, this seems to be a deeper functional change.

Previously, the second arg looks like it was _always_ set to 0. Now it appears that we're passing in the amount (decimal converted).

Was the previous code wrong in setting this field to a 0 constant? Or is the new code doing something more subtle that I'm failing to understand?

Reply:
For (hopefully) full context: 
Setting this to 0 was incorrect. This function is called only when trying to bridge the `AZERO` token. When we set this to zero, the aliased L1 sender address will receive `amountToBridge`, i.e. amount+bridge fees, and then call our EOA with 0 value. It then _only_ refunds excess bridge gas costs to the L2 target. This means that `amount` is stuck in the aliased address on L2 (for example, we can see this happened with the aliased [dev wallet](https://evm-explorer.alephzero.org/address/0xddC021f002Eaf230c9fa810B34837a3739B7208c) and the aliased [adapter](https://evm-explorer.alephzero.org/address/0xaBA092A830a5cb89a3816E3d267cB7791C16c15E)). While it's not loss of funds, since we effectively have control over the aliased address by nudging the bridge from L1, we obviously didn't want this.

The first change was to just put in `amount` as the `l2CallValue` field here, which works for `AZERO`, since `AZERO` happens to have 18 decimals, but for a token which has non-18 decimal precision, we would be passing in a value with native token precision, when the contracts [assume](https://github.com/OffchainLabs/nitro-contracts/blob/fbbcef09c95f69decabaced3da683f987902f3e2/src/bridge/AbsInbox.sol#L238) `l2Callvalue` is in 18 decimal precision. This lead to the second change, seen above, where we convert `l2CallValue` from native token decimals to 18.